packages/edit-site/src/components/layout/router.js

```diff
@@ -140,10 +148,9 @@ export default function useLayoutAreas() {
 		};
 	}
 
-	/* Patterns and Template Parts
-	 * `/wp_template_part/all` path is no longer used, but uses Patterns page screens for
-	 * backwards compatibility.
-	 */
+	// Patterns
+	// `/wp_template_part/all` path is no longer used and redirects to
+	// Patterns page for backward compatibility.
 	if ( path === '/patterns' || path === '/wp_template_part/all' ) {
```

Comment:
I was thinking that I could remove ` || path === '/wp_template_part/all'` from the conditional statement here, but since path is still `'/wp_template_part/all'` at this point, I couldn't remove it.

Reply:
I think for all the redirects we should return "null" or empty areas or something instead of trying to map like here.

Reply:
Currently the combination of `Layout` and `useLayoutAreas` always renders something "new" on every location change. But with redirects, we no longer want that. If we navigate to `/wp_template_part/all`, we want to keep the previously rendered UI, whatever it was, and just schedule an effect that will change the location to `/patterns`. It's only when reacting to the location changing  to `/patterns` when the Patterns UI is rendered.

We can achieve this by storing the `useLayoutAreas` result into state, and treating the `null` return value as "do nothing":
```js
function useLayoutAreas() {
  const path = useRouterPath();
  useEffect( () => {
    // redirect if path is /wp_template_part/all
  }, [ path ] );
  if ( path === '/wp_template_part/all' ) {
    return null;
  }
  return areasByPath( path );
}

function Layout() {
  const newAreas = useLayoutAreas();
  const [ areas, setAreas ] = useState( newAreas ?? defaultAreas );
  if ( newAreas && newAreas !== areas ) {
    setAreas( newAreas );
  }
  const { ... } = areas;
  // continue rendering areas
}
```
This way, navigating to `/wp_template_part/all` will keep the existing `areas` rendered, it won't replace them with some placeholder, it won't unmount the entire React UI only to immediately mount it again.

I'm not sure whether we should implement this all in this PR, or in a followup one.

Reply:
I'm working on implementing the code presented, but the `useRouterPath()` hook, `areasByPath()` function, `defaultAreas` variable, etc. don't exist yet. I would appreciate some more advice if you would like to include this approach in this PR 🙇

Reply:
Does this mean that I need to make further changes to our current PR? Sorry if I didn't understand your intention.